[WIP] Updating developer information with docs building instructions (#1582)
sourmash-bio/sourmash: doc/developer.md

To update your Python dependencies to the latest required for sourmash, you can run `pip install -r requirements.txt`.

> ## Installing Sourmash in the editable mode

Reviewer: ```suggestion
## Installing sourmash in the editable mode
```

> In order to use Sourmash in the command line interface or the API while developing, you will need to install it in your enviornment in editable mode by running

Reviewer: ```suggestion
In order to use sourmash in the command line interface or the API while developing, you will need to install it in your environment in editable mode by running
```

```

> pip install -e .

Reviewer 2: ```suggestion
pip install -e ".[test]"
```

ref https://github.com/dib-lab/sourmash/issues/1482

Reviewer 2: actually, I'm not certain if you can do this with just this command, or if you need both...

Author: @luizirber can you please post your recommended set of commands using tox?

Reviewer: I use `tox -e docs` for building the Python docs, and I don't use editable mode. If needed, then I use
```tox -e dev```
and follow the instructions for activating the new environment (usually
```source .tox/dev/bin/activate```
but can be 
```conda activate .tox/dev```
for people using `tox-conda`)

[1]: https://codecov.io/gh/dib-lab/sourmash/
[2]: https://github.com/dib-lab/sourmash/actions

## Building docs

> We are using [Sphinx](https://www.sphinx-doc.org/en/master/) for Sourmash main docs hosted on [readthedocs](https://sourmash.bio). For Rust, we use [rustdoc](https://doc.rust-lang.org/rustdoc/what-is-rustdoc.html).

Reviewer: ```suggestion
We are using [Sphinx](https://www.sphinx-doc.org/en/master/) for sourmash main docs hosted on [readthedocs](https://sourmash.bio). For Rust, we use [rustdoc](https://doc.rust-lang.org/rustdoc/what-is-rustdoc.html).
```

> To build the Sourmash docs run the following command. Then, inside the `html_docs` docs you will find an `index.html` file for the docs homepage.

Reviewer: ```suggestion
To build the sourmash docs run the following command. Then, inside the `html_docs` docs you will find an `index.html` file for the docs homepage.
```

```
sphinx-build -b html doc html_docs
```

> For building Rust docs you will need to change the current working directory to `sourmash/src/core` then run the following command. The docs will be generated in `sourmash/target/docs/Sourmash`

Reviewer: ```suggestion
For building Rust docs you will need to change the current working directory to `sourmash/src/core` then run the following command. The docs will be generated in `sourmash/target/docs/sourmash`
```

```
cargo doc --no-deps # --no-deps for excluding the 3rd party libs' docs from being built.
``` 

For interactive docs building we use [`cargo-watch`](https://crates.io/crates/cargo-watch). Open a terminal and run `cargo watch -x  "doc --no-deps"`, you will need to keep this command running while changing the docs. Refresh the browser for rendering the changes.

.. warning::

>     Although Rust is the backbone of Sourmash, it does not have all the functionalities that are provided in the main [Sourmash docs](https://sourmash.bio).

Reviewer: ```suggestion
    Although Rust is the backbone of sourmash, it does not have all the functionalities that are provided in the Python [sourmash API](https://sourmash.bio).
```
